src/main/kotlin/net/casual/arcade/border/MovingCenterBorderState.kt

```diff
@@ -0,0 +1,54 @@
+package net.casual.arcade.border
+
+import net.minecraft.util.Mth
+import net.minecraft.world.level.border.BorderChangeListener
+import org.joml.Vector2d
+
+class MovingCenterBorderState(
+    private val border: ArcadeBorder,
+    realDuration: Long,
+): CenterBorderState {
+    private var tickDuration = realDuration / 50.0
+    private var ticks = 0
+
+    private var center = Vector2d(0.0,0.0)
+    private var targetCenter = Vector2d(0.0,0.0)
```

Comment:
`center` and `targetCenter` should be passed in the constructor and be final (`val`)